python/mxnet/io/_im2rec.py

```diff
@@ -0,0 +1,218 @@
+    """
+    shared object manager
+    """
+    pass
+SharedObjectManager.register("PriorityQueue", queue.PriorityQueue)
+
+def _read_list(list_file, batch_size):
+    """
+    Helper function that reads the .lst file, binds it in
+    a generator and returns a batched version of the generator.
+    Parameters
+    ----------
+    list_file: str
+    input list file.
+    batch_size: int
+    batch size of the generator
+    Returns
+    -------
+    item iterator
+    iterator that contains information in .lst file
+    """
+    with open(list_file, 'r') as input_file:
+        while True:
+            fetch_data = list(itertools.islice(input_file, batch_size))
+            if not fetch_data:
+                break
+            batch = []
+            for line in fetch_data:
+                line = [i.strip() for i in line.strip().split('\t')]
+                line_len = len(line)
+                # check the data format of .lst file
+                if line_len < 3:
+                    logging.info("lst should have at least has three parts, \
+                        but only has %s parts for %s", line_len, line)
+                    continue
+                try:
+                    item = [int(line[0])] + [line[-1]] + [float(i) for i in line[1:-1]]
+                    batch.append(item)
+                except IOError:
+                    logging.info('Parsing lst met error for %s : ', line)
+                    continue
+            yield batch
+
+def _read_worker(q_out, transformer, color, quality, encoding, pass_through, data_record):
+    """
+    Helper function that will be run by the read workers
+    to fetch the image from the input queue apply
+    transformations and put it into output priority queue.
+    Parameters
+    ----------
+    q_out : priority queue
+        priority queue
+    transformer : transformer object
+        transformer object
+    color : int
+        color
+    quality : int
+        quality
+    encoding : str
+        encoding
+    pass_through: bool
+        skip encoding while packing the image
+    data_record : tuple
+        image instance to work on.
+    """
+    i, item = data_record
+    img_path = item[1]
+    try:
+        # construct the header of the record
+        header = recordio.IRHeader(0, item[2:], item[0], 0)
+        if pass_through:
+            with open(img_path, 'rb') as f_im:
+                img = f_im.read()
+            packed_image = recordio.pack(header, img)
+            q_out.put((i, packed_image, item))
+        else:
+            img = cv2.imread(img_path, color)
+            if img is None:
+                logging.info('Read a blank image for the file: %s', img_path)
+                return
+            img = transformer(array(img))
+            packed_image = recordio.pack_img(header, img, quality=quality, img_fmt=encoding)
+            q_out.put((i, packed_image, item))
+    except IOError:
+        logging.info('pack_img error on file: %s', img_path)
```

Comment:
logging.error?

Reply:
i want this to be at "info" level. because it is not an error until the number of errors crosses the user specified limit.